src/main/java/net/neoforged/neoforge/event/EventHooks.java

```diff
@@ -594,6 +595,22 @@ public static void onExplosionDetonate(Level level, Explosion explosion, List<En
         NeoForge.EVENT_BUS.post(new ExplosionEvent.Detonate(level, explosion, list));
     }
 
+    /**
+     * To be called when an explosion has calculated the knockback velocity
+     * but has not yet added the knockback to the entity caught in blast.
+     *
+     * @param level             The level that the explosion is in
+     * @param explosion         Explosion that is happening
+     * @param entity            The entity caught in the explosion's blast
+     * @param knockbackVelocity The explosion calculated velocity for the entity
+     * @return The new explosion velocity to add to the entity's existing velocity
+     */
+    public static Vec3 onExplosionKnockback(Level level, Explosion explosion, Entity entity, Vec3 knockbackVelocity) {
```

Comment:
Name this method something that reflects the input parameters and the return values, probably `getExplosionKnockback`.  It _might_ be worth renaming the `knockbackVelocity` parameter to `initialVelocity` to clarify that it is the pre-modification value, though I'm not sure how much impact that will have.